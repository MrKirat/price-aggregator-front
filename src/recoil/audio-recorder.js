import { atom } from 'recoil';

const isAudioRecorderEnabledState = atom({
    key: 'isAudioRecorderEnabledState',
    default: false
});

export default isAudioRecorderEnabledState;