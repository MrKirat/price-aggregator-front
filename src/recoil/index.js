export { default as isAudioRecorderEnabledState } from './audio-recorder';
export { default as isCameraEnabledState } from './camera';
export { default as searchStringState } from './search';
export { default as productListState } from './products';
export { default as isLoadingState } from './loading';