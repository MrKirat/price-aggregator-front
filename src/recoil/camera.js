import { atom } from 'recoil';

const isCameraEnabledState = atom({
    key: 'isCameraEnabledState',
    default: false
});

export default isCameraEnabledState;