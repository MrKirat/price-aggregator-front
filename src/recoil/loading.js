import { atom } from 'recoil';

const isLoadingState = atom({
    key: 'isLoadingState',
    default: false
});

export default isLoadingState;