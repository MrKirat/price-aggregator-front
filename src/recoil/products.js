import { atom } from 'recoil';

const productListState = atom({
    key: 'productListState',
    default: []
});

export default productListState;