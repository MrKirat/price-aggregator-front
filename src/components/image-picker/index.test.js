import React from 'react';

import { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import ImagePicker from './index';
import { RecoilRoot } from 'recoil';

it('renders without crashing', () => {
  const output = mount(<RecoilRoot><ImagePicker /></RecoilRoot>);
  expect(shallowToJson(output)).toMatchSnapshot();
});

it('renders with ImagePicker', () => {
  const output = mount(<RecoilRoot><ImagePicker /></RecoilRoot>);
  expect(output.find('ImagePicker')).toHaveLength(1);
});

it('renders with input', () => {
  const output = mount(<RecoilRoot><ImagePicker /></RecoilRoot>);
  expect(output.find('input')).toHaveLength(1);
});

it('renders with icon', () => {
  const output = mount(<RecoilRoot><ImagePicker /></RecoilRoot>);
  expect(output.find('svg')).toHaveLength(1);
});