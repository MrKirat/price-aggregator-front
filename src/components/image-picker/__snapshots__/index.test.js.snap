// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<RecoilRoot>
  <Batcher
    setNotifyBatcherOfChange={[Function]}
  />
  <ImagePicker>
    <label
      htmlFor="imagePicker"
    >
      <ForwardRef
        className="makeStyles-icon-2"
      >
        <WithStyles(ForwardRef(SvgIcon))
          className="makeStyles-icon-2"
        >
          <ForwardRef(SvgIcon)
            className="makeStyles-icon-2"
            classes={
              Object {
                "colorAction": "MuiSvgIcon-colorAction",
                "colorDisabled": "MuiSvgIcon-colorDisabled",
                "colorError": "MuiSvgIcon-colorError",
                "colorPrimary": "MuiSvgIcon-colorPrimary",
                "colorSecondary": "MuiSvgIcon-colorSecondary",
                "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                "root": "MuiSvgIcon-root",
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root makeStyles-icon-2"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
              />
            </svg>
          </ForwardRef(SvgIcon)>
        </WithStyles(ForwardRef(SvgIcon))>
      </ForwardRef>
    </label>
    <input
      accept=".jpg, .jpeg, .png"
      className="makeStyles-input-1"
      id="imagePicker"
      name="image"
      onChange={[Function]}
      type="file"
    />
  </ImagePicker>
</RecoilRoot>
`;
